BridgeAppSDK/SBAConsentDocumentFactory.swift

```diff
@@ -134,21 +124,34 @@ open class SBAConsentDocumentFactory: SBASurveyFactory {
     */
     open func reconsentStep() -> SBASubtaskStep {
         // Strip out the registration steps
-        let steps = self.steps?.filter({ !($0 is SBARegistrationStep) && !($0 is ORKRegistrationStep) })
+        let steps = self.steps?.filter({ !isRegistrationStep($0) })
         let task = SBANavigableOrderedTask(identifier: SBAOnboardingSectionBaseType.consent.rawValue, steps: steps)
         return SBASubtaskStep(subtask: task)
     }
     
     /**
+     * Return subtask step with only the steps required for consent or reconsent on login
+     */
+    open func loginConsentStep() -> SBASubtaskStep {
+        // Strip out the registration steps
+        let steps = self.steps?.filter({ !isRegistrationStep($0) })
+        let task = SBANavigableOrderedTask(identifier: SBAOnboardingSectionBaseType.consent.rawValue, steps: steps)
+        return SBAConsentSubtaskStep(subtask: task)
+    }
+    
+    private func isRegistrationStep(_ step: ORKStep) -> Bool {
+        return (step is SBARegistrationStep) || (step is ORKRegistrationStep) || (step is SBAExternalIDStep)
```

Comment:
`is` is the Swift equivalent of `isKindOfClass:` and no, `SBARegistrationStep` is *not* a subclass of `ORKRegistrationStep`. `SBARegistrationStep` inherits from `ORKFormStep`. 
`ORKRegistrationStep` overrides the `formItems` to return separate line for given/family name and does not allow for customization such as a field for external ID or data group handling. Additionally, the `SBARegistrationStep` automatically instantiates a subclass of `ORKFormStep` that handles registration in a way that is consistent with login, login via external ID, re-consent and email verification.